diff --git a/rpcs3/Emu/Cell/SPULLVMRecompiler.cpp b/rpcs3/Emu/Cell/SPULLVMRecompiler.cpp
index 6446cd0..aa676d9 100644
--- a/rpcs3/Emu/Cell/SPULLVMRecompiler.cpp
+++ b/rpcs3/Emu/Cell/SPULLVMRecompiler.cpp
@@ -717,6 +717,7 @@ class spu_llvm_recompiler : public spu_recompiler_base, public cpu_translator
 
 		const auto x = m_ir->CreateZExt(val, get_type<u64[4]>());
 
+#ifdef ARCH_X64
 		// Use integer operations here so LLVM can fold the masks into VPTERNLOG
 		if (m_use_avx512)
 		{
@@ -726,6 +727,7 @@ class spu_llvm_recompiler : public spu_recompiler_base, public cpu_translator
 			const auto e = m_ir->CreateAnd(val, 0x7f800000);
 			return uint64_as_double(m_ir->CreateSelect(m_ir->CreateIsNotNull(e), f, s));
 		}
+#endif
 
 		const auto s = m_ir->CreateShl(m_ir->CreateAnd(x, 0x80000000), 32);
 		const auto a = m_ir->CreateAnd(x, 0x7fffffff);
@@ -740,6 +742,7 @@ class spu_llvm_recompiler : public spu_recompiler_base, public cpu_translator
 	{
 		ensure(val && val->getType() == get_type<f64[4]>());
 
+#ifdef ARCH_X64
 		// Use integer operations here so LLVM can fold the masks into VPTERNLOG
 		if (m_use_avx512)
 		{
@@ -752,6 +755,7 @@ class spu_llvm_recompiler : public spu_recompiler_base, public cpu_translator
 			const auto r = m_ir->CreateOr(c, m_ir->CreateAnd(d, 0x8000000000000000));
 			return uint64_as_double(m_ir->CreateSelect(n, r, splat<u64[4]>(0).eval(m_ir)));
 		}
+#endif
 
 		const auto smax = uint64_as_double(splat<u64[4]>(0x47ffffffe0000000).eval(m_ir));
 		const auto smin = uint64_as_double(splat<u64[4]>(0x3810000000000000).eval(m_ir));
@@ -1846,6 +1850,7 @@ public:
 			[[maybe_unused]] u32 elements;
 			[[maybe_unused]] u32 dwords;
 
+#ifdef ARCH_X64
 			if (m_use_avx512)
 			{
 				stride = 64;
@@ -1859,6 +1864,7 @@ public:
 				dwords = 4;
 			}
 			else
+#endif
 			{
 				stride = 16;
 				elements = 4;
@@ -5337,6 +5343,9 @@ public:
 					}
 					}
 
+					u32 stride = 16;
+
+#ifdef ARCH_X64
 					// Check if the LS address is constant and 256 bit aligned
 					u64 clsa = umax;
 
@@ -5345,13 +5354,12 @@ public:
 						clsa = ci->getZExtValue();
 					}
 
-					u32 stride = 16;
-
 					if (m_use_avx && csize >= 32 && !(clsa % 32))
 					{
 						vtype = get_type<u8[32]>();
 						stride = 32;
 					}
+#endif
 
 					if (csize > 0 && csize <= 16)
 					{
@@ -6277,20 +6285,24 @@ public:
 				const auto sc = rotqby_reverse_base();
 				const auto sh = sc + (splat_scalar(b) >> 3);
 
+#ifdef ARCH_X64
 				if (m_use_avx512_icl)
 				{
 					return eval(vpermb(as, sh));
 				}
+#endif
 
 				return eval(pshufb_for_x86_and_tbl_for_aarch64(as, (sh & 0xf)));
 			}
 			const auto sc = rotqby_forward_base();
 			const auto sh = sc - (splat_scalar(b) >> 3);
 
+#ifdef ARCH_X64
 			if (m_use_avx512_icl)
 			{
 				return eval(vpermb(a, sh));
 			}
+#endif
 
 			return eval(pshufb_for_x86_and_tbl_for_aarch64(a, (sh & 0xf)));
 		});
@@ -6504,11 +6516,13 @@ public:
 			const auto sc = rotqby_reverse_base();
 			const auto sh = eval(sc + splat_scalar(b));
 
+#ifdef ARCH_X64
 			if (m_use_avx512_icl)
 			{
 				set_vr(op.rt, vpermb(as, sh));
 				return;
 			}
+#endif
 
 			set_vr(op.rt, pshufb_for_x86_and_tbl_for_aarch64(as, (sh & 0xf)));
 			return;
@@ -6517,11 +6531,13 @@ public:
 		const auto sc = rotqby_forward_base();
 		const auto sh = eval(sc - splat_scalar(b));
 
+#ifdef ARCH_X64
 		if (m_use_avx512_icl)
 		{
 			set_vr(op.rt, vpermb(a, sh));
 			return;
 		}
+#endif
 
 		set_vr(op.rt, pshufb_for_x86_and_tbl_for_aarch64(a, (sh & 0xf)));
 	}
@@ -6716,6 +6732,7 @@ public:
 
 	void SUMB(spu_opcode_t op)
 	{
+#ifdef ARCH_X64
 		if (m_use_avx512)
 		{
 			const auto [a, b] = get_vrs<u8[16]>(op.ra, op.rb);
@@ -6732,6 +6749,7 @@ public:
 			set_vr(op.rt, shuffle2(ax, bx, 0, 9, 2, 11, 4, 13, 6, 15));
 			return;
 		}
+#endif
 
 #ifdef ARCH_ARM64
 		if (m_use_dotprod)
@@ -7951,6 +7969,7 @@ public:
 	{
 		const auto known = known_opt.value_or(get_known_fp_class<3>(v, llvm::FPClassTest::fcNan | llvm::FPClassTest::fcInf));
 
+#ifdef ARCH_X64
 		// Avoid pessimation when full clamping isn't needed
 		if (m_use_avx512 && !(known.isKnownNeverNaN() && (known.isKnownNeverPosInfinity() || known.isKnownNeverNegInfinity())))
 		{
@@ -7958,6 +7977,7 @@ public:
 			// Normally doesn't cause issues as SNaN frequently gets quieted beforehand
 			return eval(vrangeps(v, fsplat<f32[4]>(std::bit_cast<f32, u32>(0x7f7fffff)), 0x2, 0xff));
 		}
+#endif
 
 		return eval(clamp_positive_smax(clamp_negative_smax(v, known), known));
 	}
@@ -7991,6 +8011,7 @@ public:
 			const auto a_sign = (a & splat<u32[4]>(0x80000000));
 			value_t<u32[4]> final_result = eval(splat<u32[4]>(0));
 
+#ifdef ARCH_X64
 			if (m_use_avx512)
 			{
 				value_t<u32[16]> lo_lut;
@@ -8001,6 +8022,7 @@ public:
 				final_result = vperm2d128From512(lo_lut, a_fraction, hi_lut);
 			}
 			else
+#endif
 			{
 				for (u32 i = 0; i < 4; i++)
 				{
@@ -8651,6 +8673,7 @@ public:
 					return eval(select(fcmp_uno(b != fsplat<f32[4]>(0.)), normal_fma, c));
 				}
 
+#ifdef ARCH_X64
 				// Same number of operations well preventing a serial predicate chain pessimization
 				if (m_use_avx512)
 				{
@@ -8659,6 +8682,7 @@ public:
 					const auto cb = vfixupimmps(b, a, splat<u32[4]>(0x00000800u), 0, 0xff);
 					return fma32x4(ca, cb, c, a_known, b_known);
 				}
+#endif
 
 				const auto normal_fma = fma32x4(a, b, c, a_known, b_known);
 				const auto a_cmp = fcmp_uno(a != fsplat<f32[4]>(0.));
@@ -8671,6 +8695,7 @@ public:
 			}
 		});
 
+#ifdef ARCH_X64
 		if (m_use_avx512)
 		{
 			register_intrinsic("spu_re_acc", [&](llvm::CallInst* ci)
@@ -8684,6 +8709,7 @@ public:
 			});
 		}
 		else
+#endif
 		{
 			register_intrinsic("spu_re_acc", [&](llvm::CallInst* ci)
 			{
@@ -9028,6 +9054,7 @@ public:
 				const auto a_sign = (a & splat<u32[4]>(0x80000000));
 				value_t<u32[4]> b = eval(splat<u32[4]>(0));
 
+#ifdef ARCH_X64
 				if (m_use_avx512)
 				{
 					value_t<u32[16]> lo_lut;
@@ -9038,6 +9065,7 @@ public:
 					b = vperm2d128From512(lo_lut, a_fraction, hi_lut);
 				}
 				else
+#endif
 				{
 					for (u32 i = 0; i < 4; i++)
 					{
@@ -9277,6 +9305,7 @@ public:
 
 			value_t<s32[4]> r;
 
+#ifdef ARCH_X64
 			if (m_use_avx512)
 			{
 				const auto sc = eval(bitcast<f32[4]>(max(bitcast<s32[4]>(a),splat<s32[4]>(0x0))));
@@ -9284,6 +9313,7 @@ public:
 				set_vr(op.rt, r);
 				return;
 			}
+#endif
 
 			r.value = m_ir->CreateFPToUI(a.value, get_type<s32[4]>());
 			set_vr(op.rt, select(bitcast<s32[4]>(a) > splat<s32[4]>(((32 + 127) << 23) - 1), splat<s32[4]>(-1), r & ~(bitcast<s32[4]>(a) >> 31)));
